addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -0,0 +1,406 @@
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_repr
+from pytz import timezone
+from requests.exceptions import RequestException
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    # -------------------------------------------------------------------------
+    # OVERRIDES & EXTENSIONS
+    # -------------------------------------------------------------------------
+
+    def _is_required_for_invoice(self, invoice):
+        # EXTENDS account_edi
+        if self.code != 'es_tbai':
+            return super()._is_required_for_invoice(invoice)
+
+        # TODO for Bizkaia, move_type in ('in_invoice', 'in_refund') also yields True
+        return invoice.l10n_es_tbai_is_required and invoice.move_type in ('out_invoice', 'out_refund')
+
+    def _needs_web_services(self):
+        # EXTENDS account_edi
+        return self.code == 'es_tbai' or super()._needs_web_services()
+
+    def _is_compatible_with_journal(self, journal):
+        # EXTENDS account_edi
+        if self.code != 'es_tbai':
+            return super()._is_compatible_with_journal(journal)
+
+        return journal.country_code == 'ES'
+
+    def _get_invoice_edi_content(self, invoice):
+        # EXTENDS account_edi
+        if self.code != 'es_tbai':
+            return super()._get_invoice_edi_content(invoice)
+        return self._get_l10n_es_tbai_invoice_xml(invoice)
+
+    def _check_move_configuration(self, invoice):
+        # EXTENDS account_edi
+        if self.code != 'es_tbai':
+            return super()._check_move_configuration(invoice)
+        # Ensure a certificate is available.
+        certificate = invoice.company_id.l10n_es_tbai_certificate_id
+        if not certificate:
+            return {
+                'error': _("Please configure the certificate for TicketBAI."),
+                'blocking_level': 'error',
+            }
+
+        # Ensure a tax agency is available.
+        tax_agency = invoice.company_id.mapped('l10n_es_tbai_tax_agency')[0]
+        if not tax_agency:
+            return {
+                'error': _("Please specify a tax agency on your company for TicketBAI."),
+                'blocking_level': 'error',
+            }
+
+        return None
+
+    def _post_invoice_edi(self, invoice):
+        # EXTENDS account_edi
+        if self.code != 'es_tbai':
+            return super()._post_invoice_edi(invoice)
+
+        # Configuration check
+        error_dict = self._check_move_configuration(invoice)
+        if error_dict is not None:
+            return {invoice: error_dict}
+
+        # Chain integrity check: chain head must have been REALLY posted (not timeout'ed)
+        chain_head = invoice.company_id.get_l10n_es_tbai_last_posted_invoice()
+        if chain_head and chain_head != invoice and not chain_head.l10n_es_tbai_is_in_chain:
+            raise UserError(f"TicketBAI: Cannot post invoice while chain head ({chain_head.name}) has not been posted")
```

Comment:
@odoostan Why do we raise here? (as in a cron that would not be seen)

Reply:
Aren't these errors shown in the banner of the invoice ?

Reply:
@odoostan Not that I know of

Reply:
We raise just in case the user clicks on "Process now" and expects the invoice to be processed when in fact it won't until they go to the head chain and fix potential issues preventing it from being processed first.